src/ru/desu/src/eu/kanade/tachiyomi/extension/ru/desu/Desu.kt

```diff
@@ -62,33 +62,8 @@ class Desu : ConfigurableSource, HttpSource() {
             .rateLimitHost(baseUrl.toHttpUrl(), 3)
             .build()
 
-    private fun mangaPageFromJSON(jsonStr: String, next: Boolean): MangasPage {
-        val mangaList = json.parseToJsonElement(jsonStr).jsonArray
-            .map {
-                SManga.create().apply {
-                    mangaFromJSON(it.jsonObject, false)
-                }
-            }
-
-        return MangasPage(mangaList, next)
-    }
-
-    private fun SManga.mangaFromJSON(obj: JsonObject, chapter: Boolean) {
-        val id = obj["id"]!!.jsonPrimitive.int
-
-        url = "/$id"
-        title = if (isEng.equals("rus")) {
-            obj["russian"]!!.jsonPrimitive.content
-                .split(" / ")
-                .first()
-        } else {
-            obj["name"]!!.jsonPrimitive.content
-                .split(" / ")
-                .first()
-        }
-        thumbnail_url = obj["image"]!!.jsonObject["original"]!!.jsonPrimitive.content
-
-        val ratingValue = obj["score"]!!.jsonPrimitive.floatOrNull ?: 0F
+    private fun MangaDetDto.toSManga(genresStr: String? = ""): SManga {
+        val ratingValue = score!!
```

Comment:
consider not making them nullable in the dto, since you are using non null assertion anyway

Reply:
I meant removing the nullability from the dtos (`val something: String?` => `val something: String`)
you can do it in a future pull request